### ISSUE TITLE
Broken app label: VMWare Fusion

### ISSUE THREAD

**Issue by midiman1000** [Installomator/Installomator#1946]
**What's wrong with the app label?**
The script analysis of this app, states that it finds 13.6.0, but wants to quit and install 13.5.1 instead. Which is not updating the application correctly.
[vmwarefusion.txt](https://github.com/user-attachments/files/17296358/vmwarefusion.txt)

**Installomator version**
10.6

**Additional context**
N/A

**Installomator log**

2024-10-08 09:24:01 : REQ   : vmwarefusion : ################## Start Installomator v. 10.6, date 2024-08-30
2024-10-08 09:24:01 : INFO  : vmwarefusion : ################## Version: 10.6
2024-10-08 09:24:01 : INFO  : vmwarefusion : ################## Date: 2024-08-30
2024-10-08 09:24:01 : INFO  : vmwarefusion : ################## vmwarefusion
2024-10-08 09:24:02 : INFO  : vmwarefusion : setting variable from argument DEBUG=0
2024-10-08 09:24:02 : INFO  : vmwarefusion : BLOCKING_PROCESS_ACTION=tell_user
2024-10-08 09:24:02 : INFO  : vmwarefusion : NOTIFY=success
2024-10-08 09:24:02 : INFO  : vmwarefusion : LOGGING=INFO
2024-10-08 09:24:02 : INFO  : vmwarefusion : LOGO=/System/Applications/App Store.app/Contents/Resources/AppIcon.icns
2024-10-08 09:24:02 : INFO  : vmwarefusion : Label type: dmg
2024-10-08 09:24:02 : INFO  : vmwarefusion : archiveName: VMware Fusion.dmg
2024-10-08 09:24:02 : INFO  : vmwarefusion : no blocking processes defined, using VMware Fusion as default
2024-10-08 09:24:02 : INFO  : vmwarefusion : App(s) found: /Applications/VMware Fusion.app
2024-10-08 09:24:02 : INFO  : vmwarefusion : found app at /Applications/VMware Fusion.app, version 13.6.0, on versionKey CFBundleShortVersionString
2024-10-08 09:24:02 : INFO  : vmwarefusion : appversion: 13.6.0
2024-10-08 09:24:02 : INFO  : vmwarefusion : Latest version of VMware Fusion is 13.5.1 <<<<<< ------------------------------------This is incorrect....!!!!!!
2024-10-08 09:24:02 : REQ   : vmwarefusion : Downloading https://www.vmware.com/go/getfusion to VMware Fusion.dmg
2024-10-08 09:24:02 : ERROR : vmwarefusion : error downloading https://www.vmware.com/go/getfusion
2024-10-08 09:24:02 : ERROR : vmwarefusion : File list: 
2024-10-08 09:24:02 : ERROR : vmwarefusion : File type: VMware Fusion.dmg: cannot open `VMware Fusion.dmg' (No such file or directory)
2024-10-08 09:24:02 : INFO  : vmwarefusion : Installomator did not close any apps, so no need to reopen any apps.
2024-10-08 09:24:02 : ERROR : vmwarefusion : ERROR: Error downloading https://www.vmware.com/go/getfusion error:
* Host www.vmware.com:443 was resolved.
* IPv6: 2606:4700:4400::6812:21b5, 2606:4700:4400::ac40:9a4b
* IPv4: 104.18.33.181, 172.64.154.75
*   Trying [2606:4700:4400::6812:21b5]:443...
* Connected to www.vmware.com (2606:4700:4400::6812:21b5) port 443
* ALPN: curl offers h2,http/1.1
* (304) (OUT), TLS handshake, Client hello (1):
} [319 bytes data]
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* (304) (IN), TLS handshake, Server hello (2):
{ [122 bytes data]
* (304) (IN), TLS handshake, Unknown (8):
{ [19 bytes data]
* (304) (IN), TLS handshake, Certificate (11):
{ [3400 bytes data]
* (304) (IN), TLS handshake, CERT verify (15):
{ [264 bytes data]
* (304) (IN), TLS handshake, Finished (20):
{ [36 bytes data]
* (304) (OUT), TLS handshake, Finished (20):
} [36 bytes data]
* SSL connection using TLSv1.3 / AEAD-CHACHA20-POLY1305-SHA256 / [blank] / UNDEF
* ALPN: server accepted h2
* Server certificate:
*  subject: C=US; ST=California; L=Palo Alto; O=Broadcom Inc.; CN=www.vmware.com
*  start date: Jul 26 00:00:00 2024 GMT
*  expire date: Jul 26 23:59:59 2025 GMT
*  subjectAltName: host "www.vmware.com" matched cert's "www.vmware.com"
*  issuer: C=US; O=DigiCert Inc; CN=DigiCert Global G2 TLS RSA SHA256 2020 CA1
*  SSL certificate verify ok.
* using HTTP/2
* [HTTP/2] [1] OPENED stream for https://www.vmware.com/go/getfusion
* [HTTP/2] [1] [:method: GET]
* [HTTP/2] [1] [:scheme: https]
* [HTTP/2] [1] [:authority: www.vmware.com]
* [HTTP/2] [1] [:path: /go/getfusion]
* [HTTP/2] [1] [accept: */*]
* [HTTP/2] [1] [accept-encoding: gzip, deflate]
* [HTTP/2] [1] [user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.1 Safari/605.1.15]
> GET /go/getfusion HTTP/2
> Host: www.vmware.com
> Accept: */*
> Accept-Encoding: gzip, deflate
> Connection: keep-alive
> User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.1 Safari/605.1.15
> 
* Request completely sent off
< HTTP/2 301 
< date: Tue, 08 Oct 2024 16:24:02 GMT
< content-type: text/plain; charset=utf-8
< content-length: 124
< location: https://download3.vmware.com/software/FUS-1351/VMware-Fusion-13.5.1-23298085_universal.dmg
< cf-ray: 8cf77a978cf16bae-DFW
< cf-cache-status: HIT
< age: 30366
< cache-control: no-cache, no-store, must-revalidate, proxy-revalidate, max-age=0
< last-modified: Fri, 12 Jul 2024 15:12:43 GMT
< strict-transport-security: max-age=31536000; includeSubDomains
< vary: Accept, Accept-Encoding
< bc_locale: en-us
< bclang: en-us
< bcsrc: bcvmware
< content-security-policy: frame-ancestors 'self' esbroadcom.lookbookhq.com mfbroadcom.lookbookhq.com; script-src 'self' data: blob: https://script.crazyegg.com https://www.google-analytics.com https://www.googletagmanager.com https://cdn.cookielaw.org https://geolocation.onetrust.com https://ajax.googleapis.com https://snap.licdn.com https://searchunify.com https://searchunify.broadcom.com https://tag.demandbase.com/9a4d64cf696797e4.min.js https://tag.demandbase.com https://connect.facebook.net https://vjs.zencdn.net https://www.gstatic.com *.adroll.com *.eloqua.com *.en25.com *.bluekai.com *.oraclecloud.com *.brightcove.com *.brightcove.net https://images.sw.broadcom.com 'nonce-brcm229OD7z5eg5hA1voZew9TQ=='; object-src 'self';
< x-content-type-options: nosniff
< x-frame-options: SAMEORIGIN
< x-permitted-cross-domain-policies: master-only
< x-xss-protection: 1; mode=block
< set-cookie: __cf_bm=cKREijpyAyn1..cOSwR5mdbqNuIqett7wdEggZIQKfQ-1728404642-1.0.1.1-fPh9W_yHpMEZ145WuVenf4NsYXP9z7Vuy4zPiOQD__v_Rpjx8qMNHU9iF0v.skQEx1SldNp_vp6SdZNz57ab3A; path=/; expires=Tue, 08-Oct-24 16:54:02 GMT; domain=.www.vmware.com; HttpOnly; Secure; SameSite=None
< server: cloudflare
< 
* Ignoring the response-body
* Connection #0 to host www.vmware.com left intact
* Issue another request to this URL: 'https://download3.vmware.com/software/FUS-1351/VMware-Fusion-13.5.1-23298085_universal.dmg'
* Host download3.vmware.com:443 was resolved.
* IPv6: 2606:4700:7::a5, 2a06:98c1:58::a5
* IPv4: 162.159.140.167, 172.66.0.165
*   Trying [2606:4700:7::a5]:443...
* Connected to download3.vmware.com (2606:4700:7::a5) port 443
* ALPN: curl offers h2,http/1.1
* (304) (OUT), TLS handshake, Client hello (1):
} [325 bytes data]
* (304) (IN), TLS handshake, Server hello (2):
{ [122 bytes data]
* (304) (IN), TLS handshake, Unknown (8):
{ [19 bytes data]
* (304) (IN), TLS handshake, Certificate (11):
{ [3228 bytes data]
* (304) (IN), TLS handshake, CERT verify (15):
{ [264 bytes data]
* (304) (IN), TLS handshake, Finished (20):
{ [36 bytes data]
* (304) (OUT), TLS handshake, Finished (20):
} [36 bytes data]
* SSL connection using TLSv1.3 / AEAD-CHACHA20-POLY1305-SHA256 / [blank] / UNDEF
* ALPN: server accepted h2
* Server certificate:
*  subject: C=US; ST=California; L=Palo Alto; O=Broadcom Inc.; CN=hostupdate.vmware.com
*  start date: May  6 00:00:00 2024 GMT
*  expire date: May  7 23:59:59 2025 GMT
*  subjectAltName: host "download3.vmware.com" matched cert's "download3.vmware.com"
*  issuer: C=US; O=DigiCert Inc; CN=DigiCert TLS RSA SHA256 2020 CA1
*  SSL certificate verify ok.
* using HTTP/2
* [HTTP/2] [1] OPENED stream for https://download3.vmware.com/software/FUS-1351/VMware-Fusion-13.5.1-23298085_universal.dmg
* [HTTP/2] [1] [:method: GET]
* [HTTP/2] [1] [:scheme: https]
* [HTTP/2] [1] [:authority: download3.vmware.com]
* [HTTP/2] [1] [:path: /software/FUS-1351/VMware-Fusion-13.5.1-23298085_universal.dmg]
* [HTTP/2] [1] [accept: */*]
* [HTTP/2] [1] [accept-encoding: gzip, deflate]
* [HTTP/2] [1] [user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.1 Safari/605.1.15]
> GET /software/FUS-1351/VMware-Fusion-13.5.1-23298085_universal.dmg HTTP/2
> Host: download3.vmware.com
> Accept: */*
> Accept-Encoding: gzip, deflate
> Connection: keep-alive
> User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.1 Safari/605.1.15
> 
* Request completely sent off
< HTTP/2 522 
< date: Tue, 08 Oct 2024 16:24:02 GMT
< content-type: text/html; charset=UTF-8
< content-length: 7137
< vary: Accept-Encoding
< strict-transport-security: max-age=31536000; includeSubDomains
< referrer-policy: same-origin
< cache-control: private, max-age=0, no-store, no-cache, must-revalidate, post-check=0, pre-check=0
< expires: Thu, 01 Jan 1970 00:00:01 GMT
< content-security-policy: *
< expect-ct: enforce, max-age=300
< x-content-type-options: nosniff
< x-frame-options: SAMEORIGIN
< x-xss-protection: 1; mode=block
< server: cloudflare
< cf-ray: 8cf77a9879fb4623-DFW
< 
* The requested URL returned error: 522
* process_pending_input: nghttp2_session_mem_recv() returned -902:The user callback function failed
* Connection #1 to host download3.vmware.com left intact
curl: (56) The requested URL returned error: 522

2024-10-08 09:24:02 : REQ   : vmwarefusion : ################## End Installomator, exit code 2

**Comment by uurazzle** [comment]
Hi @midiman1000 :

Thanks for the feedback on this issue.

Just wanted to flag that this issue has gotten worse. The `softwareupdate.vmware.com` URL that was introduced to fix an earlier breakage is now itself broken — Broadcom has since redirected it to `support.broadcom.com`, which doesn't serve the DMG.

This appears to be the same infrastructure change described here: https://www.bleepingcomputer.com/news/software/vmware-workstation-auto-updates-broken-after-broadcom-url-redirect/

Notably, this isn't just an Installomator problem — in-app auto-update in VMware Fusion/Workstation is also broken for the same reason, so there's no vendor-side workaround to point users to at the moment.

The label will need a new `downloadURL` that either points to a stable Broadcom CDN path or scrapes the Broadcom support portal for the current release. Happy to help test once a working URL is identified.

So, for the time being probably best we close this issue, since there appears to be nothing Installomator can solve.

**Comment by dannykorpan** [comment]
Would that be legal?
https://archive.org/download/vmwareworkstationarchive/Fusion/

**Comment by uurazzle** [comment]
HI @dannykorpan:

I am not a lawyer, but try to play one on Github ;-) 

Using older versions of VMware Fusion (often called "abandonware" when hosted on sites like the Internet Archive) falls into a legal gray area where it is technically a violation of copyright, but practically carries almost zero risk for individual users. I would recommend verifying with your organization's legal department.

However, recent changes by Broadcom (who now owns VMware) have made the latest versions of Fusion completely free, which might make using an "old" version unnecessary.

**Comment by dannykorpan** [comment]
@uurazzle Thanks! The real question now is whether I should submit a pull request for this with a link to archive.org.
